### PR TITLE
fix: Merge MyProfileEditForm and MyProfileEditFormModal

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Bugfix**

### Description

This PR fixes a lot of issues that came with https://github.com/artsy/eigen/pull/6158 and https://github.com/artsy/eigen/pull/6076 being worked on at the same time. Because we renamed `MyProfileEditFormModal` to `MyProfileEditForm` all the changes made in https://github.com/artsy/eigen/pull/6076 were lost in the refactoring work from https://github.com/artsy/eigen/pull/6158.

### PR Checklist (tick all before merging)

- [ ] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [ ] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [ ] I have documented any follow-up work that this PR will require, or it does not require any.
- [ ] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [ ] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [x] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

### CODE REVIEW

File: src/lib/AppRegistry.tsx [hunk 1]
```diff
@@ -380,9 +380,13 @@ export const modules = defineModules({
...
   MyCollectionArtworkFullDetails: reactModule(MyCollectionArtworkFullDetailsQueryRenderer),
-  MyProfile: reactModule(MyProfile, {
-    isRootViewForTabName: "profile",
-  }),
+  MyProfile: reactModule(
+    MyProfile,
+    {
+      isRootViewForTabName: "profile",
+    },
+    MyProfileHeaderMyCollectionAndSavedWorksScreenQuery
```
Comment (olerichter00) [inline src/lib/AppRegistry.tsx:388]: Bringing the query back for prefetching.

File: src/lib/Scenes/MyAccount/updateMyUserProfile.ts [hunk 2]
```diff
@@ -16,6 +16,8 @@ export const updateMyUserProfile = async (input: UpdateMyProfileInput = {}) => {
               email
               name
               phone
+              profession
+              otherRelevantPositions
```
Comment (olerichter00) [inline src/lib/Scenes/MyAccount/updateMyUserProfile.ts:20]: We need to include these fields in the mutation to update them in the Relay store after updating the profile.

File: src/lib/Scenes/MyProfile/MyProfileHeaderMyCollectionAndSavedWorks.tsx [hunk 3]
```diff
@@ -167,8 +167,7 @@ export const MyProfileHeaderMyCollectionAndSavedWorksFragmentContainer = createR
   MyProfileHeaderMyCollectionAndSavedWorks,
   {
     me: graphql`
-      fragment MyProfileHeaderMyCollectionAndSavedWorks_me on Me
-      @argumentDefinitions(enableCollectorProfile: { type: "Boolean", defaultValue: false }) {
```
[the remark this answers is not in the document]
Comment (olerichter00) [reply src/lib/Scenes/MyProfile/MyProfileHeaderMyCollectionAndSavedWorks.tsx:171]: The tests are still passing and the content is displayed correctly depending on whether the feature is enabled or not.

[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/lib/Scenes/MyProfile/MyProfileHeaderMyCollectionAndSavedWorks.tsx:171]: no need to have it, it was an overkill anyway

**Review by MounirDhahri — DISMISSED** [review]
🚢🚢🚢🚢

File: src/lib/AppRegistry.tsx (hunk 1 above)
Comment (rajsam003) [reply src/lib/AppRegistry.tsx:388]: This was the reason 💡

File: src/lib/Scenes/MyProfile/MyProfileEditForm.tsx [hunk 4]
```diff
@@ -325,14 +415,14 @@ const LoadingSkeleton = () => {
         <PlaceholderBox width={99} height={99} borderRadius={50} />
         <PlaceholderText width={100} height={20} marginTop={5} marginLeft={20} />
       </Flex>
-      {Array(enableCollectorProfile ? 4 : 1).fill(
-        <Flex mt={30}>
+      {[...Array(enableCollectorProfile ? 4 : 1)].map((_x, i) => (
```
[the remark this answers is not in the document]
Comment (rajsam003) [reply src/lib/Scenes/MyProfile/MyProfileEditForm.tsx:418]: Ah ok

**Review by rajsam003 — APPROVED** [review]
Thanks for the changes Ole 🙏

**Pull request merged**